virtcontainers/qemu.go

```diff
@@ -1275,6 +1275,59 @@ func (q *qemu) hotplugVhostUserDevice(vAttr *config.VhostUserDeviceAttrs, op ope
 	return nil
 }
 
+func (q *qemu) qomGetSlot(qomPath string) (uint8, error) {
+	addr, err := q.qmpMonitorCh.qmp.ExecQomGet(q.qmpMonitorCh.ctx, qomPath, "addr")
+	if err != nil {
+		return 0xff, err
+	}
+	addrf, ok := addr.(float64)
+	// XXX going via float makes no real sense, but that's how
+	// JSON works, and we should get away with it for the small
+	// values we have here
+	if !ok {
```

Comment:
@dgibson - I've just looked at the qemu code and it *appears* that the returned address is actually a JSON string, not a JSON number (aka float). Please could you check that?

Reply:
I'm pretty sure it's a number - you may be looking at the `legacy_addr` property rather than `addr`.  I don't think the things I'm trying with this could possibly work if that were not the case.  But I will double check.